### ISSUE TITLE
issue installing

### ISSUE THREAD

**Issue by Greatguy2** [cryptoeax/arbbot#157]
I have done like it was in manual and when i try to start it give me this.didnt had any witdraws. it is clean account

pc@pc-VirtualBox:/var/www/arbbot$ php main.php
14:53:31: ARBITRATOR V2.0 launching...
sh: 1: /usr/sbin/sendmail: not found
14:53:31: Loading config...
14:53:31: Enabling Bittrex...

PHP Fatal error:  Class Bittrex contains 1 abstract method and must therefore be declared abstract or implement the remaining methods (Exchange::detectDuplicateWithdrawals) in /var/www/arbbot/bot/xchange/Bittrex.php on line 6

**Comment by Scorpic1** [comment]
Have same error.

**Comment by Greatguy2** [comment]
Who know how to solve it?

**Comment by sblackstein** [comment]
The temporary way, which is how I solved it, is to comment out arbbot/bot/Exchange.php, line 297 i.e.:

```
  // public abstract function detectDuplicateWithdrawals();
```

so when the method isn't implemented in extending classes, it will skip the error until they get implemented.

**Comment by Greatguy2** [comment]
@sblackstein 
Great it helped, but now i have other issue :) Cant connect to poloniex

09:04:34: ARBITRATOR V2.0 launching...
sh: 1: /usr/sbin/sendmail: not found
09:04:34: Loading config...
09:04:34: Enabling Bittrex...
09:04:34: Enabling Bleutrade...
09:04:34: Enabling Poloniex...
09:04:34: Configured 3 exchanges!
09:04:34: Testing exchange access...
09:04:35: BITTREX [OK]
09:04:39: BLEUTRADE [OK]
09:04:40: [POLONIEX] Could not get reply (HTTP 403):
09:04:40: [POLONIEX] Could not get reply (HTTP 403):
09:04:40: [POLONIEX] Could not get reply (HTTP 403):
09:04:40: [POLONIEX] Could not get reply (HTTP 403):
09:04:40: [POLONIEX] Could not get reply (HTTP 403):
09:04:40: POLONIEX [ERROR]
[POLONIEX] Could not get reply (HTTP 403):

**Comment by sblackstein** [comment]
Oh, hmm. So, Poloniex only lets you use their API to allow withdrawals if you set your source IP in the configuration. Has your IP changed maybe, and that's why you're getting a 403 Forbidden?

**Comment by Greatguy2** [comment]
No, i have changed api, allowed only my ip adress. enabled 2 factor. but is is still showing  error 403. maybe you have any ideas?

**Comment by jjc6676** [comment]
Just to be sure, you did update the key and secret in the config file correct?

**Comment by Greatguy2** [comment]
@jjc6676 Yes, and double checked. And checked my ip. I have static IP address.

**Comment by heiko7even** [comment]
i also get this error ...

**Comment by dmeledon** [comment]
It looks like when @cryptoeax merged in pending_deposits not everything that got lost was restored. detectDuplicateWithdrawals() appears to have gotten removed from the bot entirely. I had to comment out line 435 on CoinManage.php where it calls this function during domanage() as well.
